Modules/Module_Processing/Module_ASL/xASL_wrp_ProcessM0.m

```diff
@@ -146,7 +146,7 @@ function xASL_wrp_ProcessM0(x)
 
 if isfield(x.modules.asl, 'bRegisterM02ASL') && ~x.modules.asl.bRegisterM02ASL
     fprintf('M0 registration (to ASL or T1w) is skipped upon request\n');
-elseif ~strcmpi(x.Q.M0, 'UseControlAsM0') && isempty(regexpi(x.Q.Sequence, 'spiral'))
+elseif ~strcmpi(x.Q.M0, 'UseControlAsM0') && isempty(regexpi(x.Q.PulseSequenceType, 'spiral', 'once'))

```

Comment:
So this has to be x.Q.Readout -> '3d_spiral' 

(note that we had this, assuming that x.Q.Sequence could only contain '2d_epi, 3d_grase, 3d_spiral, which we created ourselves. Now it can have 2d_spiral)

Reply:
The point is that in the old version, we used a parameter created by ourselves (x.Q.Sequence), which could exclusively be 2d_epi, 3d_grase, 3d_spiral. Now, we have replaced this by a BIDS parameter that has multiple options. So we changed the functionality (in general, not in the most common cases).

So we can only replace checking x.Q.Sequence by checking both MRAcquisition & PulseSequenceType

Reply:
Yes, I understand. And I can add the 3D check here. But my point was that it's not really necessary here.

Reply:
So everywhere (not only here) where you renamed x.Q.Sequence to *PulseSequenceType, you potentially need to add the MRAcquisitionType check (even though you think it won't matter). That is my only wish ;)